x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/create_new_index_pattern.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getHttp } from '../../../kibana_services';
+import { CreateDocSourceResp, INDEX_SOURCE_API_PATH } from '../../../../common';
+
+export const createNewIndexAndPattern = async (indexName: string) => {
+  return await getHttp().fetch<CreateDocSourceResp>({
+    path: `/${INDEX_SOURCE_API_PATH}`,
+    method: 'POST',
+    body: convertObjectToBlob({
```

Comment:
How about just using JSON.stringify(obj) instead of blob. Checkout all the http calls in file_upload they just use SON.stringify(obj) and work fine

Reply:
you're right, this is my mistake. `BodyInit` allows for a string 

```type BodyInit = Blob | BufferSource | FormData | URLSearchParams | ReadableStream<Uint8Array> | string;```

apologies for the noise and the the run-around.